src/factorizations/diagonal.jl

```diff
@@ -95,15 +96,18 @@ end
 
 # f_vals
 # ------
-for f! in (:eig_vals!, :eigh_vals!, :svd_vals!)
+# this is invalid for svd_vals since we cannot sort between blocks
+for f! in (:eig_vals!, :eigh_vals!)
```

Comment:
This might be a bit risky for `eig(h)` as well. For example, `eigh_full` returns a sorted output of the eigenvalues in the LAPACK/dense case. So we could in principle also implement this for the `DiagonalAlgorithm`, but then this will also invalidate this implementation. So maybe  a block per block approach is the safest, even if there is a little bit extra overhead.